jmetal-core/src/main/java/org/uma/jmetal/util/experiment/component/GenerateEvolutionChart.java

```diff
@@ -0,0 +1,199 @@
+ *       }
+ * 
+ * <b>3. (at Algorithm) Execute indicators</b>
+ *      for example:
+ *          if(HistoricAlgorithm.testToCalculate(evaluations,maxEvaluations)){
+ *               HistoricAlgorithm.calculateIndicators(evaluations, maxEvaluations, problem.getName(), population,history);
+ *           }
+ * 
+ * <b>4. (at Experiment) Generate data files and R scripts</b>
+ *      for example:
+ *          HistoryData.printAllDataInstances(experiment, indicators);
+ *          new GenerateEvolutionChart(experiment, indicators).run();
+ *          
+ * 
+ * @author Lucas Prestes <lucas.prestes.lp@gmail.com> 
+ */
+public class GenerateEvolutionChart implements ExperimentComponent{
+    
+    Experiment<?,?> experiment;
+    List<String> indicators;
+    String outputDir;
+    
+    public GenerateEvolutionChart(Experiment experiment, List<String> indicators) {
+        this.experiment = experiment;
+        this.indicators = indicators;
+        this.outputDir = experiment.getExperimentBaseDirectory() + "/EvolutionProcess/";
+    }
+    
+    
+
+    @Override
+    public void run() throws IOException {
+        for (int i = 0; i < experiment.getProblemList().size(); i++) {
+            String problem = (experiment.getProblemList().get(i)).getTag() ;
+
+            File f = new File(outputDir);
+            if(!f.exists()){
+                f.mkdirs();
+            }
+            List<ExperimentAlgorithm<DoubleSolution, List<DoubleSolution>> > algorithmList = new ArrayList();
+            //Set algorithm of this problem
+            for (Object algorithm : experiment.getAlgorithmList()) {
+                ExperimentAlgorithm<DoubleSolution, List<DoubleSolution>> alg = (ExperimentAlgorithm<DoubleSolution, List<DoubleSolution>>) algorithm;
+                if(alg.getProblemTag().equals(problem)){
+                    algorithmList.add(alg);
+                }
+            }
+            String mainScript= "pdf(\"plot_"+problem+".pdf\")\n";
+            for (String indicator : indicators) {
+                String script = generateRscript(algorithmList, indicator, problem);
+                try (OutputStream os = new FileOutputStream(outputDir+"Rscript_chart_"+indicator+"_"+problem+".R")) {
+                    PrintStream ps = new PrintStream(os);
+                    ps.print(script);
+                    ps.close();
+                }
+                mainScript+="source(\"Rscript_chart_"+indicator+"_"+problem+".R\")\n";
+            }
+            try (OutputStream os = new FileOutputStream(outputDir+"main_"+problem+".R")) {
+                PrintStream ps = new PrintStream(os);
+                ps.print(mainScript);
+                ps.close();
+            }
+        }
+    }
+    
+    
+    
+    
+    private String generateRscript(List<ExperimentAlgorithm<DoubleSolution, List<DoubleSolution>> > algorithmList, String indicator, String problem){
+        int i = 1;
+        String str = "# Read data file\n" ;
```

Comment:
Use the `PrintStream` here rather than string concatenation. Concatenation is costly, and `PrintStream` allows you to use `println` to avoid expliciting newlines.